src/plugins/intel_npu/src/plugin/npuw/lazy_tensor.cpp

```diff
@@ -62,11 +63,21 @@ struct Const {
         NPUW_ASSERT(m_read_from_bin && "Underlying data should have been read first!");
         return m_read_from_bin;
     }
-    void read_weight(const ov::npuw::s11n::Weights& weights) {
+    void read_weight(const ov::npuw::s11n::CompiledDescDeserializeContext& ctx) {
         NPUW_ASSERT(!m_node &&
                     "LazyTensor can only read weight when it's being deserialized and not created from a Constant!");
         m_read_from_bin = ov::Tensor(m_cached_type, m_cached_shape);
-        std::memcpy(m_read_from_bin.data(), weights->get_ptr(m_offset), m_byte_size);
+
+        if (ctx.weights) {
+            std::memcpy(m_read_from_bin.data(), ctx.weights->get_ptr(m_offset), m_byte_size);
+        } else {
+            auto it = ctx.consts_cache.find({m_offset, m_byte_size});
```

Comment:
every if is a seed for a virtual method. :)

Reply:
I'd keep it in mind for the future refactoring